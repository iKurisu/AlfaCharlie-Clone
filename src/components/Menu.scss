@import 'sass/_variables.scss';

.menu {
  width: 100%;
  height: 100%;
  position: fixed;
  top: 0;
  left: 0;
  z-index: 99;
}

.menu.show {
  visibility: visible;
}

.menu.hide {
  visibility: hidden;
  transition: visibility 0s 1.85s; // 1s
}

.menu .vertical-nav {
  width: 58px;
  height: 100vh;
  position: absolute;
  top: 0;
  right: 33px;
  display: block;
}

.menu .vertical-nav-links {
  width: 100vh;
  height: 40px;
  display: flex;
  align-items: center;
  justify-content: center;
  transform-origin: 30px 50%;
}

.menu .vertical-nav-links li:last-child {
  position: absolute;
  right: 20px;
}
