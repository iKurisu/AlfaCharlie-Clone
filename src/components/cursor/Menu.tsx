import React from "react";
import { connect } from "react-redux";
import { AppState } from "store";
import Outline from "./Outline";
import Progress from "./Progress";

interface MappedState {
  menuToggled: boolean;
  hoveringElementID: number;
}

type Props = MappedState;

export const Menu = ({
  menuToggled,
  hoveringElementID
}: Props): JSX.Element => (
  <div className="menu-cursor" style={{ opacity: menuToggled ? 1 : 0 }}>
    <Outline />
    <Progress progress={hoveringElementID / 4} />
  </div>
);

const mapState = ({ menu }: AppState): MappedState => ({
  menuToggled: menu.toggled,
  hoveringElementID: menu.hoveringElementID
});

export default connect(mapState)(Menu);
