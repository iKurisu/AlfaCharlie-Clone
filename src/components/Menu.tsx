import React from "react";
import { connect } from "react-redux";
import { AppState } from "store";
import White from "./menu/White";
import Gray from "./menu/Gray";
import Navigation from "./menu/Navigation";
import VerticalNav from "./VerticalNav";
import Images from "./menu/Images";
import "./Menu.scss";

interface MappedState {
  isOpen: boolean;
}

type Props = MappedState;

export const Menu = ({ isOpen }: Props): JSX.Element => (
  <div className={`menu ${isOpen ? "show" : "hide"}`}>
    <Navigation isOpen={isOpen} />
    <Images isOpen={isOpen} />
    <VerticalNav menu={true} show={isOpen} delay={1400} />
    <White isOpen={isOpen} />
    <Gray isOpen={isOpen} />
  </div>
);

const mapState = ({ menu }: AppState): MappedState => ({
  isOpen: menu.toggled
});

export default connect(mapState)(Menu);
