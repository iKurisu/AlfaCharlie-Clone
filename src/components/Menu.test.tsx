import React from "react";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";
import { Menu } from "./Menu";

describe("menu", (): void => {
  const props = {
    isOpen: false,
    hoveringElementID: 0,
    previousElementID: 0,
    hoverElement: jest.fn()
  };

  it("renders correctly", (): void => {
    const menu = renderer.create(<Menu {...props} />);

    expect(menu).toMatchSnapshot();
  });

  it("has correct class", (): void => {
    const hiddenMenu = shallow(<Menu {...props} />);
    const visibleMenu = shallow(<Menu {...{ ...props, isOpen: true }} />);

    expect(hiddenMenu.find(".menu").hasClass("hide")).toBe(true);
    expect(visibleMenu.find(".menu").hasClass("show")).toBe(true);
  });
});
