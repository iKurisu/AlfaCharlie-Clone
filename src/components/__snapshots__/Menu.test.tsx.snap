// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="menu hide"
>
  <div
    className="menu-nav-wrapper -flex"
  >
    <a
      className="menu-home-link"
    >
      <svg
        className="menu-logo"
        viewBox="0 0 156.23 156.23"
        x="0px"
        y="0px"
      >
        <path
          d="M34.8,82.75c0.25,0.53,0.44,1.08,0.58,1.66c0.66,2.87-0.18,5.48-2.53,7.83c-0.77,0.77-1.57,1.38-2.4,1.82   c-2.84,1.47-5.5,0.97-7.96-1.49c-3.59-3.65-1.91-9.09,5.06-16.33l-2.69-2.69c-0.8-0.8-1.55-1.44-2.24-1.91   c-1.85-1.3-3.43-1.3-4.72,0c-0.39,0.39-0.66,0.8-0.83,1.24c-0.19,0.75,0.11,1.82,0.91,3.23c0.28,0.5,0.47,0.86,0.58,1.08   c0.75,1.57,0.52,2.96-0.66,4.14c-0.28,0.28-0.58,0.5-0.91,0.66c-1.3,0.75-2.49,0.58-3.56-0.5c-0.47-0.47-0.79-1.04-0.95-1.7   c-0.66-2.6,0.88-5.77,4.64-9.53c0.97-0.97,1.96-1.77,2.98-2.4c4.01-2.4,7.74-1.88,11.19,1.57l8.66,8.66   c1.33,1.33,2.32,1.71,2.98,1.16c0.77-0.83,0.29-2.28-1.45-4.35l1.12-1.04c0.77,0.94,1.35,1.77,1.74,2.49   c0.97,1.96,0.44,3.95-1.57,5.97C40.41,84.41,37.76,84.55,34.8,82.75z M33.76,87.52c0.72-1.6,0.37-3.63-1.04-6.09l-4.02-4.02   c-1.19,1.3-2.06,2.47-2.61,3.52c-0.94,2.05-0.5,3.98,1.33,5.8c2.43,2.38,4.36,2.9,5.8,1.57C33.42,88.11,33.6,87.85,33.76,87.52z"
        />
        <path
          d="M66.67,26.43c-0.08-3.73,1.34-7.06,4.27-9.99c2.15-2.15,4.38-3.52,6.67-4.1c2.38-0.55,4.17-0.22,5.39,0.99   c0.22,0.22,0.41,0.47,0.58,0.75c0.72,1.38,0.37,2.78-1.04,4.19c-0.66,0.66-1.33,1.05-1.99,1.16c-0.86,0.14-2-0.35-3.44-1.45   c-0.61-0.44-1.05-0.75-1.33-0.91c-1.41-0.86-2.72-0.68-3.94,0.54c-0.61,0.61-1.01,1.26-1.2,1.95c-0.58,2.35,1.45,5.84,6.09,10.48   c2.32,2.32,4.35,3.88,6.09,4.68c2.35,1.08,4.47,0.66,6.38-1.24c0.72-0.72,1.33-1.55,1.82-2.49c1.27-2.43,1.46-4.86,0.58-7.29   l1.37-0.79c0.44,0.99,0.7,2.06,0.79,3.19c0.19,3.29-1.15,6.37-4.02,9.24c-1.85,1.85-3.84,3.15-5.97,3.9   c-4.59,1.55-8.89,0.3-12.93-3.73C68.15,32.8,66.75,29.77,66.67,26.43z"
        />
        <path
          d="M121.43,73.49c-0.25-0.53-0.44-1.08-0.58-1.66c-0.66-2.87,0.18-5.48,2.53-7.83c0.77-0.77,1.57-1.38,2.4-1.82   c2.84-1.47,5.5-0.97,7.96,1.49c3.59,3.65,1.91,9.09-5.06,16.33l2.69,2.69c0.8,0.8,1.55,1.44,2.24,1.91c1.85,1.3,3.43,1.3,4.72,0   c0.39-0.39,0.66-0.8,0.83-1.24c0.19-0.75-0.11-1.82-0.91-3.23c-0.28-0.5-0.47-0.86-0.58-1.08c-0.75-1.57-0.52-2.96,0.66-4.14   c0.28-0.28,0.58-0.5,0.91-0.66c1.3-0.75,2.49-0.58,3.56,0.5c0.47,0.47,0.79,1.04,0.95,1.7c0.66,2.6-0.88,5.77-4.64,9.53   c-0.97,0.97-1.96,1.77-2.98,2.4c-4.01,2.4-7.74,1.88-11.19-1.57l-8.66-8.66c-1.33-1.33-2.32-1.71-2.98-1.16   c-0.77,0.83-0.29,2.28,1.45,4.35l-1.12,1.04c-0.77-0.94-1.35-1.77-1.74-2.49c-0.97-1.96-0.44-3.95,1.57-5.97   C115.83,71.83,118.48,71.69,121.43,73.49z M122.47,68.72c-0.72,1.6-0.37,3.63,1.04,6.09l4.02,4.02c1.19-1.3,2.06-2.47,2.61-3.52   c0.94-2.05,0.5-3.98-1.33-5.8c-2.43-2.38-4.36-2.9-5.8-1.57C122.82,68.13,122.64,68.39,122.47,68.72z"
        />
        <path
          d="M89.56,129.8c0.08,3.73-1.34,7.06-4.27,9.99c-2.15,2.15-4.38,3.52-6.67,4.1c-2.38,0.55-4.17,0.22-5.39-0.99   c-0.22-0.22-0.41-0.47-0.58-0.75c-0.72-1.38-0.37-2.78,1.04-4.19c0.66-0.66,1.33-1.05,1.99-1.16c0.86-0.14,2,0.35,3.44,1.45   c0.61,0.44,1.05,0.75,1.33,0.91c1.41,0.86,2.72,0.68,3.94-0.54c0.61-0.61,1.01-1.26,1.2-1.95c0.58-2.35-1.45-5.84-6.09-10.48   c-2.32-2.32-4.35-3.88-6.09-4.68c-2.35-1.08-4.47-0.66-6.38,1.24c-0.72,0.72-1.33,1.55-1.82,2.49c-1.27,2.43-1.46,4.86-0.58,7.29   l-1.37,0.79c-0.44-0.99-0.7-2.06-0.79-3.19c-0.19-3.29,1.15-6.37,4.02-9.24c1.85-1.85,3.84-3.15,5.97-3.9   c4.59-1.55,8.89-0.3,12.93,3.73C88.08,123.44,89.48,126.46,89.56,129.8z"
        />
        <g
          className="menu-logo-lines"
        >
          <line
            x1="28.39"
            x2="127.84"
            y1="28.39"
            y2="127.84"
          />
          <line
            x1="28.39"
            x2="127.84"
            y1="127.84"
            y2="28.39"
          />
        </g>
      </svg>
    </a>
    <nav
      className="menu-nav-links -flex"
    >
      <a
        className="menu-nav-link"
      >
        <span
          className="link-characters"
        >
          <span
            className="link-character"
          >
            a
          </span>
          <span
            className="link-character"
          >
            g
          </span>
          <span
            className="link-character"
          >
            e
          </span>
          <span
            className="link-character"
          >
            n
          </span>
          <span
            className="link-character"
          >
            c
          </span>
          <span
            className="link-character"
          >
            y
          </span>
        </span>
      </a>
      <a
        className="menu-nav-link"
      >
        <span
          className="link-characters"
        >
          <span
            className="link-character"
          >
            w
          </span>
          <span
            className="link-character"
          >
            o
          </span>
          <span
            className="link-character"
          >
            r
          </span>
          <span
            className="link-character"
          >
            k
          </span>
        </span>
      </a>
      <a
        className="menu-nav-link"
      >
        <span
          className="link-characters"
        >
          <span
            className="link-character"
          >
            j
          </span>
          <span
            className="link-character"
          >
            o
          </span>
          <span
            className="link-character"
          >
            u
          </span>
          <span
            className="link-character"
          >
            r
          </span>
          <span
            className="link-character"
          >
            n
          </span>
          <span
            className="link-character"
          >
            a
          </span>
          <span
            className="link-character"
          >
            l
          </span>
        </span>
      </a>
      <a
        className="menu-nav-link"
      >
        <span
          className="link-characters"
        >
          <span
            className="link-character"
          >
            c
          </span>
          <span
            className="link-character"
          >
            o
          </span>
          <span
            className="link-character"
          >
            n
          </span>
          <span
            className="link-character"
          >
            t
          </span>
          <span
            className="link-character"
          >
            a
          </span>
          <span
            className="link-character"
          >
            c
          </span>
          <span
            className="link-character"
          >
            t
          </span>
        </span>
      </a>
    </nav>
  </div>
  <div
    className="menu-white"
    style={
      Object {
        "opacity": 1,
        "transform": "translateX(100%)",
      }
    }
  />
  <div
    className="menu-gray"
    style={
      Object {
        "transform": "translateX(-100%)",
      }
    }
  />
</div>
`;
